test/Chainweb/Test/Mempool.hs

```diff
@@ -102,15 +106,22 @@ arbitraryGasPrice :: Gen GasPrice
 arbitraryGasPrice = GasPrice . ParsedDecimal . abs <$> arbitraryDecimal
 
 instance Arbitrary MockTx where
-  arbitrary = MockTx <$> chooseAny
-                        <*> arbitraryGasPrice
-                        <*> pure mockBlockGasLimit
-                        <*> pure emptyMeta
+  arbitrary = MockTx
+      <$> chooseAny
+      <*> arbitraryGasPrice
+      <*> pure mockBlockGasLimit
+      <*> pure emptyMeta
     where
       emptyMeta = TransactionMetadata zero Time.maxTime
       zero = Time.Time (Time.TimeSpan (Time.Micros 0))
 
-type InsertCheck = MVar (Vector MockTx -> IO (Vector (Maybe InsertError)))
+type BatchCheck =
+    Vector (TransactionHash, MockTx)
+    -> IO (V.Vector (Either (TransactionHash, InsertError) (TransactionHash, MockTx)))
+
+-- TODO Why is this inside an MVar?
+type InsertCheck = MVar BatchCheck
```

Comment:
Remove TODO and add this explanation. We have an MVar so tests can override/modify the instance's insert check for tests. To make quickcheck testing not take _forever_ for remote we run the server and client mempools in a resource pool -- the test pulls an already running instance from the pool so we need a way to twiddle its check function for the tests.